feat: allow symfony 7

Chris53897 opened the pull request:
First try to support symfony 7.
I guess the problem is a bug with the virtual_request_stack which should already be fixed  https://github.com/symfony/symfony/commit/99a0dba420137e42ad2ad7ce7cd8b18d4f8edf79.
But i still get the error.

`Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The service "debug.log_processor" has a dependency on a non-existent service ".virtual_request_stack".`

Chris8934 committed:
feat: allow symfony 7 (WIP)

Chris8934 committed:
feat: allow symfony 7 (WIP)

Chris8934 committed:
feat: allow symfony 7 (WIP)

Chris8934 committed:
feat: allow symfony 7 (WIP)

alexislefebvre commented:
I suggest to wait for the stable release of Symfony before working on this PR, using `dev` versions may bring more issues than expected.

Chris53897 commented:
This is a good idea.
I just want to test my own bundle for symfony 7, that uses this bundle.

Chris53897 commented:
This PR could probably fix it. See referenced Issue for more Details.
https://github.com/symfony/symfony/pull/52604

Update: 7.0 RC2 did fixed this.

Chris8934 committed:
feat: allow symfony 7 (WIP)

Chris8934 committed:
feat: allow symfony 7 (WIP)

Chris8934 committed:
feat: allow symfony 7 (WIP)

Chris8934 committed:
feat: allow symfony 7

Chris53897 commented:
@alexislefebvre Can you please review?

There is a failing (optional?) run.
Symfony ^7.0 on PHP 8.2 flags remove --dev --no-progress doctrine/phpcr-bundle doctrine/phpcr-odm
Your requirements could not be resolved to an installable set of packages.

alexislefebvre commented:
`phpcr/phpcr-utils` only supports Symfony up to 6, see https://github.com/phpcr/phpcr-utils/blob/1.8.0/composer.json#L32

I don't understand why the CI is green when one step failed.

alexisLefebvre commented on .github/workflows/tests.yml:
```diff
@@ -31,6 +31,10 @@ jobs:
           - php-version: 8.2
             # Remove doctrine/phpcr-* in order to allow doctrine/persistence v3
             composer-flags: "remove --dev --no-progress doctrine/phpcr-bundle doctrine/phpcr-odm"
+          - php-version: 8.2
+            symfony-version: "^7.0"
+            # Remove doctrine/phpcr-* in order to allow doctrine/persistence v3
+            composer-flags: "remove --dev --no-progress doctrine/phpcr-bundle doctrine/phpcr-odm"
```
```suggestion
```

alexisLefebvre commented on composer.json:
```diff
@@ -74,5 +74,7 @@
             "dev-master": "2.x-dev"
         }
-    }
+    },
+    "minimum-stability": "dev",
+    "prefer-stable": true
```
```suggestion
    }
```

Chris53897 committed:
Update composer.json

Co-authored-by: Alexis Lefebvre <alexislefebvre@gmail.com>

Chris53897 committed:
Update .github/workflows/tests.yml

Co-authored-by: Alexis Lefebvre <alexislefebvre@gmail.com>

Chris53897 commented:
There is already a PR at `phpcr/phpcr-utils` for Symfony 7 support.
https://github.com/phpcr/phpcr-utils/pull/214

Chris53897 commented:
I did commit your suggestions. But now there is no testrun for symfony 7.

alexislefebvre commented:
My bad, I misread the `composer-flags: "remove --dev …` part. Please restore it. :bow:

Chris8934 committed:
feat: allow symfony 7

Chris53897 commented:
I reverted that change. CI fails becauseof minimum-stability, i removed in the other suggested commit.
Lets wait for a stable release of symfony and the other linked PR is merged.
It is not worth the effort to change the CI for 3 days.

alexislefebvre commented:
Thanks! Here is the last release: https://github.com/liip/LiipTestFixturesBundle/releases/tag/2.7.0

Chris53897 commented:
Thanks
